### PR TITLE
VPD-1775 initial refactoring

### PR DESCRIPTION
**Pull request by ian-hmrc**
Initial work to simplify the calculation in preparation to deal with inconsistent User Answers data.

### CODE REVIEW

File: test/services/returns/BuildReturnSubmissionServiceSpec.scala [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+          .set(EnterDutySuspensePage, dutySuspenseVolumes).success.value
+          .set(DeclarationPage, declaration).success.value
+
+        val returnCreateRequest = service.buildSubmission(userAnswers, obligation, vpdId, Map(PeriodKey(obligation.periodKey) -> 1050))
+
+        returnCreateRequest.vapingProductsProduced.vapingProdManufactured mustBe yes
+        returnCreateRequest.vapingProductsProduced.returns.size mustBe 1
+        returnCreateRequest.vapingProductsProduced.returns.head mustBe RegularReturn("641", 105.0, 1, 10500)
+
+        //        returnCreateRequest.spoiltProduct.get.spoiltProductFilled mustBe no
```
Comment (ghousedev) [inline test/services/returns/BuildReturnSubmissionServiceSpec.scala]: DId you want to remove this comment?  Plus a few others in this file.

Comment (ian-hmrc) [reply test/services/returns/BuildReturnSubmissionServiceSpec.scala]: I was going to wait for the next PR for this one, but I've done that. It required setting-up more authentic user answers test data, so I've done that by starting from a completely nil return, and then allowed the tests to override the answers required for the particular scenario. That's allowed the code to be updated to test the spoilt in each test.

I feel like we're repeating assertions in the tests right now. I suspect these will get narrower as I add more cases for the broken user answers. Likely in the next PR.

**Review by ghousedev — DISMISSED** [review]
👍

**Review by ghousedev — APPROVED** [review]
👍
